src/aiProviders/continue/listTablesContextProvider.ts

```diff
@@ -0,0 +1,127 @@
+import { ContextItem, ContextProviderDescription, ContextProviderExtras, ContextSubmenuItem, IContextProvider, LoadSubmenuItemsArgs } from "@continuedev/core";
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as vscode from "vscode";
+import { JobManager } from "../../config";
+import { JobInfo } from "../../connection/manager";
+import Schemas from "../../database/schemas";
+import Table from "../../database/table";
+import { findPossibleTables } from "../context";
+
+const listDb2Table: ContextProviderDescription = {
+  title: "list Db2i Tables",
+  displayTitle: "Db2i-tables",
+  description: "Add Db2i Table info to Context",
+  type: "submenu"
+}
+
+export let listDb2TableContextProvider: Boolean = false;
+
+class ListDb2iTables implements IContextProvider {
```

Comment:
Add a constructor here and let it have a `schema` parameter and store in a class member. Then remove `getDefaultSchema` and use the new member instead.